### ISSUE TITLE
Customized source code formatting with styler

### ISSUE THREAD

**Issue by lorenzwalthert** [mlr-org/mlr#2278]
Thanks for this package, I really like it (have only scratched the surface) and in particular, the documentation is great.

I am the maintainer of the styler package, which is a flexible source code formatter for R code. We have implemented the [tidyverse style guide](http://style.tidyverse.org) and a joke style guide called [oneliner](https://github.com/lorenzwalthert/oneliner) to demonstrate that styler is capable of implementing a wide range of code formatting rules.

I just read your wiki that links to [a coding style](https://github.com/rdatsci/PackagesInfo/wiki/R-Style-Guide) which seems to correspond to the tidyverse style guide to a large extend. With my knowledge as one of the main developer of styler, I think it would not be hard to configure styler so it implements the aforementioned style guide.

It could then be used to make sure existing code complies with the style guide. If you'd be interested in that, I can get us started.
cc: @krlmlr

**Comment by lorenzwalthert** [comment]
styler cannot cover all aspects of the style guide obviously, but it can cover the pure code formatting aspects of it.

**Comment by mb706** [comment]
Instead of forking into a new package you might consider making `styler` parametric (possibly with a config file in the same dir, the package root, or the user's home dir) to lower the total maintenance burden.

**Comment by krlmlr** [comment]
@mb706: Forking helps determine what needs to be configurable in the fist place, I think the idea is to reconcile later into a more configurable version of *styler*.

**Comment by lorenzwalthert** [comment]
@mb706 styler is pretty flexible, it's just that almost none of these functions are exported. I mean I outlined my ideas for [extending styler](https://github.com/lorenzwalthert/oneliner/blob/master/vignettes/extending-styler.Rmd) earlier but since the overlap with the tidyverse style guide is quite large and we don't have the `stylerinfra` package yet that exports all these utilities we need, maybe it is best to start work on that within the existing package structure anyways. But in terms of unit tests ect., I think at some point we should branch out to keep styler as lightweight as possible.
oneliner was different because it only relied on a few internal functions from styler I accessed with `:::`, which we can't do if we want to submit anything to CRAN.

**Comment by pat-s** [comment]
@lorenzwalthert Can you quickly tell me how to change the indention to two spaces?
Played around a bit with a style guide for mlr that we could apply in Travis.

I found the instructions at http://styler.r-lib.org/articles/customizing_styler.html#implementation-details a bit complicated to understand regarding the indention.

**Comment by GegznaV** [comment]
Something like `styler::style_text(command, indent_by = 2)`?

**Comment by lorenzwalthert** [comment]
Sorry I did not get back to you. I created a style guide a few months ago within r-lib/styler that seemed to match the mlr style quite closely. You can find the branch [here](https://github.com/lorenzwalthert/styler/tree/mlr-style). I am still not sure how we handle third party style guides, so maybe this eventually needs to live outside styler. An example of how it works (after building the package):

``` r
library(styler)
style_text("g <- function(a,
           b = 3) {
           a + b
           }", transformers = mlr_style(strict = TRUE))
#> g = function(a,
#>   b = 3) {
#>   a + b
#> }
```

<sup>Created on 2018-11-22 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Happy to discuss with you how to go forward.

**Comment by pat-s** [comment]
> Happy to discuss with you how to go forward. I think we best apply the style guide to the whole of mlr, create a PR, check the diff and see where it does not produce satisfying results.

That's exactly what I wanted to do. In the long run, having it done by Travis would be the most convenient option. 

Maybe putting the transformer object into `.Rprofile` is the easiest way right now?
This way we can go with the dev of `styler` and do not have to rely on an (outdated) branch.

```r
<some transformer modifications>
style_pkg(transformers = style_mlr)
```

> I am still not sure how we handle third party style guides

If not in the pkg itself, it will be hard to follow the pkg dev (if custom styles are outsourced). What is the main concern about putting it into the pkg? That there will be too many styles at some point?
